src/main/java/org/ladocuploader/app/cli/TransmitterCommands.java

```diff
@@ -61,17 +69,30 @@ public TransmitterCommands(TransmissionRepository transmissionRepository,
     @ShellMethod(key = "transmit")
     public void transmit() throws IOException, JSchException, SftpException {
         log.info("Finding submissions to transmit...");
+        OffsetDateTime submittedAtCutoff = OffsetDateTime.now().minusHours(TWO_HOURS);
         for (TransmissionType transmissionType : transmissionTypes) {
-            var submissions = this.transmissionRepository.submissionsToTransmit(Sort.unsorted(), transmissionType);
-            log.info("Total submissions to transmit for {} is {}", transmissionType.name(), submissions.size());
-            if (submissions.size() > 0) {
+
+            List<Submission> queuedSubmissions = transmissionRepository.submissionsToTransmit(Sort.unsorted(), transmissionType);
+            int totalQueued = queuedSubmissions.size();
+            if (queuedSubmissions.isEmpty()) {
+                log.info("Nothing to transmit. Exiting.");
+                context.close();
+                return;
+            }
+            log.info("Found %s queued transmissions".formatted(totalQueued));
+
+            queuedSubmissions = queuedSubmissions.stream()
+                    .filter(submission -> (submission.getSubmittedAt().isBefore(submittedAtCutoff))).toList();
+            log.info("Total submissions to transmit for {} is {}", transmissionType.name(), queuedSubmissions.size());
+            if (queuedSubmissions.size() > 0) {
                 log.info("Transmitting submissions for {}", transmissionType.name());
-                transmitBatch(submissions, transmissionType);
+                transmitBatch(queuedSubmissions, transmissionType);
             } else {
                 log.info("Skipping transmission for {}", transmissionType.name());
             }
 
         }
+        context.close();
```

Comment:
Do you know how the context behaves if theres an SFTP exception or something?

Reply:
good point - I will look into that